### PR TITLE
Refactor : use stackWindow useForm change Height

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
cities old codes changes

### CODE REVIEW

File: src/pages/cities/index.js [hunk 1]
```diff
@@ -104,13 +101,26 @@ const City = () => {
...
-    setWindowOpen(true)
+    openForm()
   }
 
-  const edit = obj => {
-    setSelectedRecordId(obj.recordId)
-    setWindowOpen(true)
+  function openForm(recordId) {
+    stack({
+      Component: CityWindow,
```
Comment (NoraBitarSM) [inline src/pages/cities/index.js:114]: call the city form

**Review by NoraBitarSM — CHANGES_REQUESTED** [review]

**Comment by NoraBitarSM** [comment]
functionality tested and reviewed.

File: src/pages/cities/Forms/CityForm.js [hunk 2]
```diff
@@ -7,173 +7,157 @@ import toast from 'react-hot-toast'
...
-                  formik.setFieldValue('countryId', newValue?.recordId)
-                } else {
+          const res = await getRequest({
+            extension: SystemRepository.City.get,
+            parameters: `_recordId=${recordId}`
+          })
 
-                  formik.setFieldValue('countryId', '')
-                }
+          setInitialData(res.record)
```
Comment (RebeccaFrem02) [inline src/pages/cities/Forms/CityForm.js:80]: can't we use formik.setValues here?

File: src/pages/cities/Forms/CityForm.js [hunk 3]
```diff
@@ -7,173 +7,157 @@ import toast from 'react-hot-toast'
...
+        extension: SystemRepository.City.set,
+        record: JSON.stringify(obj)
+      })
 
-        const response = await postRequest({
-          extension: SystemRepository.City.set,
-          record: JSON.stringify(obj)
+      if (!recordId) {
+        toast.success('Record Added Successfully')
+        setInitialData({
```
Comment (RebeccaFrem02) [inline src/pages/cities/Forms/CityForm.js:58]: can't we use formik.setValues here?

File: src/pages/cities/index.js [hunk 4]
```diff
@@ -17,20 +16,16 @@ import { SystemRepository } from 'src/repositories/SystemRepository'
 import { ResourceIds } from 'src/resources/ResourceIds'
 
 // ** Windows
-import CityWindow from './Windows/CityWindow'
+import CityForm from 'src/pages/cities/Forms/CityForm'
```
Comment (RebeccaFrem02) [inline src/pages/cities/index.js:19]: the comments in the import part serve no purpose

File: src/pages/cities/Forms/CityForm.js [hunk 5]
```diff
@@ -7,173 +7,157 @@ import toast from 'react-hot-toast'
 import { RequestsContext } from 'src/providers/RequestsContext'
 import { useInvalidate } from 'src/hooks/resource'
 import { ResourceIds } from 'src/resources/ResourceIds'
+import { useForm } from 'src/hooks/form'
 
 // ** Custom Imports
```
Comment (RebeccaFrem02) [inline src/pages/cities/Forms/CityForm.js:12]: these comments serve no purpose

File: src/pages/cities/Forms/CityForm.js [hunk 6]
```diff
@@ -7,173 +7,157 @@ import toast from 'react-hot-toast'
...
 import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
 
-export default function CityForm ({ labels, recordId, maxAccess }) {
+export default function CityForm({ labels, recordId, maxAccess }) {
+  const [isLoading, setIsLoading] = useState(false)
+  const [editMode, setEditMode] = useState(!!recordId)
 
-    const [isLoading, setIsLoading] = useState(false)
-    const [editMode, setEditMode] = useState(!!recordId)
+  const [initialValues, setInitialData] = useState({
```
Comment (RebeccaFrem02) [inline src/pages/cities/Forms/CityForm.js:22]: can't you remove this state and add the initial values directly to the formik?

File: src/pages/cities/Forms/CityForm.js [hunk 7]
```diff
@@ -7,173 +7,157 @@ import toast from 'react-hot-toast'
...
-    const { getRequest, postRequest } = useContext(RequestsContext)
+  const invalidate = useInvalidate({
+    endpointId: SystemRepository.City.page
+  })
 
-    const invalidate = useInvalidate({
-      endpointId: SystemRepository.City.page
-    })
+  const { formik } = useForm({
+    initialValues,
```
Comment (RebeccaFrem02) [inline src/pages/cities/Forms/CityForm.js:39]: try moving the initial values here

File: src/pages/cities/Forms/CityForm.js (hunk 2 above)
Comment (Alinadimmokdad) [reply src/pages/cities/Forms/CityForm.js:80]: yes

File: src/pages/cities/Forms/CityForm.js (hunk 3 above)
Comment (Alinadimmokdad) [reply src/pages/cities/Forms/CityForm.js:58]: no need for both

File: src/pages/cities/Forms/CityForm.js (hunk 6 above)
Comment (Alinadimmokdad) [reply src/pages/cities/Forms/CityForm.js:22]: yes

File: src/pages/cities/Forms/CityForm.js [hunk 8]
```diff
@@ -7,173 +7,151 @@ import toast from 'react-hot-toast'
...
 
 // ** Custom Imports
 import CustomTextField from 'src/components/Inputs/CustomTextField'
 
 import { SystemRepository } from 'src/repositories/SystemRepository'
 import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
 
-export default function CityForm ({ labels, recordId, maxAccess }) {
+export default function CityForm({ labels, recordId, maxAccess }) {
+  const [isLoading, setIsLoading] = useState(false)
```
Comment (hadichahine) [inline src/pages/cities/Forms/CityForm.js:19]: Who's using this?

File: src/pages/cities/Forms/CityForm.js [hunk 9]
```diff
@@ -7,173 +7,151 @@ import toast from 'react-hot-toast'
...
+          <CustomTextField
+            name='reference'
+            label={labels.reference}
+            value={formik.values.reference}
+            required
+            maxAccess={maxAccess}
+            readOnly={editMode}
+            onChange={formik.handleChange}
+            onClear={() => formik.setFieldValue('reference', '')}
+            error={formik.touched.reference && Boolean(formik.errors.reference)}
```
Comment (hadichahine) [inline src/pages/cities/Forms/CityForm.js:94]: why pass boolean?

File: src/pages/cities/Forms/CityForm.js [hunk 10]
```diff
@@ -7,173 +7,151 @@ import toast from 'react-hot-toast'
...
+            columnsInDropDown={[
+              { key: 'reference', value: 'Reference' },
+              { key: 'name', value: 'Name' }
+            ]}
+            values={formik.values}
+            required
+            maxAccess={maxAccess}
+            onChange={(event, newValue) => {
+              formik.setFieldValue('stateId', null)
+              if (newValue) {
```
Comment (hadichahine) [inline src/pages/cities/Forms/CityForm.js:128]: why this if statement? I know it's present in code somewhere else by we can have  `formik.setFieldValue('countryId', newValue?.recordId || '')`

File: src/pages/cities/index.js [hunk 11]
```diff
@@ -134,22 +144,10 @@ const City = () => {
           maxAccess={access}
           isLoading={false}
```
Comment (hadichahine) [inline src/pages/cities/index.js:145]: Why are we passing this?

File: src/pages/cities/index.js [hunk 12]
```diff
@@ -47,7 +42,9 @@ const City = () => {
     access,
     search,
     clear,
-    refetch
+    refetch,
```
Comment (hadichahine) [inline src/pages/cities/index.js:45]: I don't think we're using it here.

**Review by hadichahine — CHANGES_REQUESTED** [review]
Some of the comments are present in multiple places in PR. Please make sure duplicates are addressed too.

**Review by hadichahine — APPROVED** [review]

**Pull request merged**